addons/mail/static/tests/discuss/core/web/messaging_menu.test.js

```diff
@@ -135,9 +133,9 @@ test("deleted message should not show parent message reference and mentions", as
     ).click();
     await click(".o-mail-Message-moreMenu .o-dropdown-item:contains(Delete)");
     await click(".o_dialog button:contains(Delete)");
-    await contains(".o-mail-Message:not(:has(.o-mail-Message-bubble.o-orange))", {
-        text: "This message has been removed",
-    });
+    await contains(
+        ".o-mail-Message:not(:has(.o-mail-Message-bubble.o-orange)):has(:text('This message has been removed'))"
```

Comment:
A bit weird for test to mix `:contains()` and `:has(:text())`.

Reply:
I didnt touch those `contains` , it's not possible to check one by one if it is a fix use of contains and :has(:text()). But I think if it works as the original logic, it should be fine?

Reply:
The problem is that `:contains()` is more permissive than `:has(:text())`, so doing positive assertion with `:has(:text())` and negative with `:contains()` is fine, but the other-way is not guarantee of non-presence of the text from `:contains()`.

So it's weird but also could potentially cause similar problem as "negative assertion without matching positive assertion" problem, so that's why I think we should change the positive assertion to `:has(:text())` or change the ones in diff by `:contains()` to be consistent.